felix/dataplane/linux/qos/qos.go

```diff
@@ -290,15 +294,49 @@ func CreateEgressQdisc(tbs *TokenBucketState, hostDeviceName string, ifbDeviceNa
 
 		err = netlink.QdiscAdd(qdisc)
 		if err != nil {
-			return fmt.Errorf("create ingress qdisc %+v on dev %s: %s", qdisc, hostDeviceName, err)
+			return fmt.Errorf("create ingress qdisc %+v on dev %s: %w", qdisc, hostDeviceName, err)
+		}
+	}
+
+	// List filters on hostDevice to clean up filters and bandwidth plugin ifb devices (those have a "bwp" prefix), if present.
+	filters, err := netlink.FilterList(hostDevice, netlink.HANDLE_INGRESS)
+	if err != nil {
+		return fmt.Errorf("list filters on dev %s: %w", hostDeviceName, err)
+	}
+	log.Debugf("Cleaning up existing filters on dev %s: %+v", hostDeviceName, filters)
+	for _, filter := range filters {
+		u32Filter, ok := filter.(*netlink.U32)
+		if ok {
```

Comment:
nit: i'd have a slight preference for `if !ok {continue} as it save indentation (2 levels in this nested for)